core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -181,7 +181,7 @@ protected void validate(TableMetadata currentMetadata, Snapshot snapshot) {}
    * Apply the update's changes to the base table metadata and return the new manifest list.
    *
    * @param metadataToUpdate the base table metadata to apply changes to
-   * @param snapshot
+   * @param snapshot parent snapshot passed to child operations.
```

Comment:
Ah thanks for adding the commenting.  Just a nit, I think this should say something like "snapshot to apply the changes to". Child operations doesn't make too much sense imo, here we're just saying "Apply the changes to this this metadata and specific snapshot".